activerecord/lib/active_record/encryption/encryptable_record.rb

```diff
@@ -0,0 +1,240 @@
+        end
+
+        private
+          def encrypt_attribute(name, key_provider: nil, key: nil, deterministic: false, downcase: false,
+                                ignore_case: false, subtype: ActiveModel::Type::String.new, previous: [], **context_properties)
+            raise Errors::Configuration, ":ignore_case can only be used with deterministic encryption" if ignore_case && !deterministic
+            raise Errors::Configuration, ":key_provider and :key can't be used simultaneously" if key_provider && key
+
+            encrypted_attributes << name.to_sym
+
+            key_provider = build_key_provider(key_provider: key_provider, key: key, deterministic: deterministic)
+
+            attribute name, :encrypted, key_provider: key_provider, downcase: downcase || ignore_case, deterministic: deterministic,
+                      subtype: subtype, previous_types: build_previous_types(previous, subtype), **context_properties
+
+            preserve_original_encrypted(name) if ignore_case
+            validate_column_size(name) if ActiveRecord::Encryption.config.validate_column_size
+            ActiveRecord::Encryption.encrypted_attribute_was_declared(self, name)
+          end
+
+          def build_previous_types(previous_config_list, type)
+            previous_config_list = [previous_config_list] unless previous_config_list.is_a?(Array)
+            previous_config_list.collect do |previous_config|
+              key_provider = build_key_provider(**previous_config.slice(:key_provider, :key, :deterministic))
+              context_properties = previous_config.slice(*ActiveRecord::Encryption::Context::PROPERTIES.without(:key_provider))
+              ActiveRecord::Encryption::EncryptedAttributeType.new \
+                  key_provider: key_provider, downcase: previous_config[:downcase] || previous_config[:ignore_case],
+                  deterministic: previous_config[:deterministic], subtype: type, **context_properties
+            end
+          end
+
+          def build_key_provider(key_provider: nil, key: nil, deterministic: false)
+            return DerivedSecretKeyProvider.new(key) if key.present?
+            return key_provider if key_provider
+
+            if deterministic && (deterministic_key = ActiveRecord::Encryption.config.deterministic_key)
+              DerivedSecretKeyProvider.new(deterministic_key)
+            end
+          end
+
+          def preserve_original_encrypted(name)
+            original_attribute_name = "#{ORIGINAL_ATTRIBUTE_PREFIX}#{name}".to_sym
+
+            if !ActiveRecord::Encryption.config.support_unencrypted_data && !column_names.include?(original_attribute_name.to_s)
+              raise Errors::Configuration, "To use :ignore_case for '#{name}' you must create an additional column named '#{original_attribute_name}'"
+            end
+
+            encrypts original_attribute_name
+
+            define_method name do
```

Comment:
When you use `ignore_case: true` you must add a column `original_<name>` where the original value is going to be stored (encrypted). In the column `<name>`, it will store an encrypted downcased version of the value. 

So it needs the regular "encrypted attribute" declaration, like any other field, since they are encrypted. But, then, it overrides the accessors to:

- Serve the "original value" when getting the value
- Keep both columns in sync when assigning a new value

I'll see about using a module there to fix the problem you mention.

Reply:
Addressed here https://github.com/basecamp/rails/commit/807c3d7f3553681dc8156c7f7461cb29856f997f